pycausal_explorer/meta/_doubleML.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+from sklearn.model_selection import cross_val_predict, train_test_split
+from sklearn.utils.validation import check_is_fitted
+
+from pycausal_explorer.base import BaseCausalModel
+
+from sklearn.base import clone
+
+
+class DoubleMLLinear(BaseCausalModel):
+    """
+    Double Machine Learning model. Estimates causal effect using two different models:
+    one models outcome, and another models treatment.
+
+    Linear version. Should be used when you believe treatment effect is linear, and
+    the treatment variable is continuous
+
+    Parameters
+    ----------
+    outcome_learner : estimator object
+        base learner to use when predicting outcome. Should implement fit and predict methods.
+    treatment_learner : estimator object
+        base learner to use when predicting treatment. Should implement fit and predict methods.
+    score : basestring
+        Which score function to use. One of "partial-out" and "orthogonal"
+    """
+
+    def __init__(
+        self, outcome_learner, treatment_learner, score="partial-out", k_fold=5
+    ):
+        valid_scores = ["partial-out", "orthogonal"]
+        if score not in valid_scores:
+            raise ValueError("Score has to be one of " + str(valid_scores))
+        self.outcome_leaner = clone(outcome_learner)
+        self.treatment_learner = clone(treatment_learner)
+
+        self.k_fold = k_fold
+        self.score = score
+        self.is_fitted_ = False
+
+    def fit(self, X, y, *, treatment):
+        self.is_fitted_ = True
```

Comment:
Better to put it at last. If there is an error before the method end we do not change this variable